Let `docker/build-push-action` handle caching

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request modifies the `build` and `build-push-all` jobs to allow the [docker/build-push-action] action to manage its own cache through the GitHub Actions cache.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

The initial motivation is that I noticed that neither of these jobs have been caching for some time now. See the following log entries for [actions/cache]:

```log
Run actions/cache@v4
Cache not found for input keys: buildx-Linux-7320dadd1527eed465e04ccb[25](https://github.com/cisagov/skeleton-docker/actions/runs/14216906822/job/39835754459#step:7:26)d6c6bb105799d7, buildx-Linux-
```

```log
Post job cleanup.
Warning: Path Validation Error: Path(s) specified in the action for caching do(es) not exist, hence no cache is being saved.
```

When I looked into options I saw that the [docker/build-push-action] had the capability to cache with GitHub Actions directly and that there are some caveats around using GitHub Actions caching locally for Docker (per the [documentation](https://docs.docker.com/build/ci/github-actions/cache/#local-cache)).
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass except for the `build-push-all` job which needs changes in #221 to pass. I verified that new cache entries are showing up using `gh cache list`.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

[actions/cache]: https://github.com/actions/cache
[docker/build-push-action]: https://github.com/docker/build-push-action

mcdonnnj committed:
Let docker/build-push-action handle caching

Since the only thing being cached in the `build` and `build-push-all`
jobs is Docker info it makes sense to let the action handle caching
since it supports caching directly with the Actions cache.

dav3r commented on .github/workflows/build.yml:
```diff
@@ -259,23 +258,14 @@ jobs:
-          cache-from: type=local,src=${{ env.BUILDX_CACHE_DIR }}
-          cache-to: type=local,dest=${{ env.BUILDX_CACHE_DIR }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
```
What do you think about adding a brief comment explaining why we prefer the `max` cache mode to `min`?

mcdonnnj replied:
Please let me know what you think of https://github.com/cisagov/skeleton-docker/pull/222/commits/677e97f725aabcf1decd67f6b588dfda75037e7c.

jsf9k replied:
Perfect!

dav3r replied:
Excellent, thanks!

dav3r reviewed (approved):
👍

mcdonnnj committed:
Explain why we use a specific setting for an action

We set the cache mode to `max` for the docker/build-push-action action
so we should explain why we are overriding the default of `min`.

Co-authored-by: dav3r <david.redmin@gwe.cisa.dhs.gov>
